### PR TITLE
Fix python scripts to work with python3

### LINKED ISSUE #1856: SPIR-V command line tests fail with python 3
Tests pass with python 2.7, but fail with python 3. Example output:
```
      Start  4: spirv-tools-test_spirv_unit_tests
 1/92 Test #85: spirv_opt_tests ......................................***Failed    0.47 sec
SPIRV tool test suite: "SpirvOptBase"
1/18       TestAssemblyFileAsOnlyParameter          -Failed-
<root>/SPIRV-Tools/build/tools/spirv-opt <root>/SPIRV-Tools/build/test/tools/opt/tmpz4a8om9u/tmphbnvkvsk.spvasm.spv -o <root>/SPIRV-Tools/build/test/tools/opt/tmpz4a8om9u/output.spv
name 'reduce' is not defined
STDOUT:
b''
STDERR:
b''
```

### LINKED ISSUE #2407: Add python to Readme build instructions
I'm following the build instructions and immediately hit with python runtime errors that I'm now diving into. The build instructions don't even mention that having python installed is a prerequisite / dependency of the build system -- can the build documentation be updated to list the requirements of the build environment?

```
8>------ Build started: Project: core_tables, Configuration: Release x64 ------
8>Building Custom Rule D:/projects/SPIRV-Tools-src/source/CMakeLists.txt
8>CMake does not need to re-run because D:\projects\SPIRV-Tools-src\build\source\CMakeFiles\generate.stamp is up-to-date.
8>Generate info tables for SPIR-V vunified1 core instructions and operands.
8>Traceback (most recent call last):
8>  File "D:/projects/SPIRV-Tools-src/utils/generate_grammar_tables.py", line 749, in <module>
8>    main()
8>  File "D:/projects/SPIRV-Tools-src/utils/generate_grammar_tables.py", line 612, in main
8>    import argparse
8>ImportError: No module named argparse
```

### PR DESCRIPTION
**Pull request by s-perron**
- Updated script to work with python3 and python2.

- Added required tools.

We added a section to the readme to mention the tools that are needed to
build and test spirv-tools.  For the compiler, the compilers used by the
bots are mentioned.

The bots have been changed.  The windows bots will not use python 3.6 for testing.  The other bots will still use python 2.7.  Both Python2 and Python3 will be tested.

Fixes #2407.
Fixes #1856.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -269,6 +269,30 @@ cmake [-G <platform-generator>] <spirv-dir>
...
 development environment.
 
+### Tools you'll need
+
+For building and testing SPIRV-Tools, the following tools should be
+installed regardless of your OS:
+
+- [CMake](http://www.cmake.org/): for generating compilation targets.  Version
+  2.8.12 or later.
+- [Python](http://www.python.org/): for utility scripts and running the test suite.
```
Comment (dnovillo) [inline README.md]: Mention that either Python 2 or 3 are allowed?

File: README.md [hunk 2]
```diff
@@ -269,6 +269,30 @@ cmake [-G <platform-generator>] <spirv-dir>
...
+
+SPIRV-Tools is regularly tests with the the following compilers:
```
Comment (dnovillo) [inline README.md]: s/regularly tests/regularly tested/

File: README.md [hunk 3]
```diff
@@ -269,6 +269,36 @@ cmake [-G <platform-generator>] <spirv-dir>
...
 
+### Tools you'll need
+
+For building and testing SPIRV-Tools, the following tools should be
+installed regardless of your OS:
+
+- [CMake](http://www.cmake.org/): for generating compilation targets.  Version
+  2.8.12 or later.
+- [Python](http://www.python.org/): for utility scripts and running the test 
+suite. Version 2 or 3.
```
Comment (dnovillo) [inline README.md]: So, if we are saying Python 2 or 3, should the bots make sure that they test with both?

Comment (s-perron) [reply README.md]: I changed the windows bots so they use python3.  Is that enough?

Comment (dnovillo) [reply README.md]: But the Linux bot is also using python3.  I was thinking that maybe we leave one bot running python2.

Comment (s-perron) [reply README.md]: Look at https://source.cloud.google.com/results/invocations/440d865b-cb1d-4da4-8581-133a2ad7cb62/targets/graphics_shader_compiler%2Fspirv-tools%2Flinux-clang-debug%2Fpresubmit/log

You will find 

> -- SPIRV-Tools: nosetests was not found - python support code will not be tested
> -- Found PythonInterp: /home/kbuilder/.pyenv/shims/python (found version "2.7.6")

Comment (s-perron) [reply README.md]: True, we did not force python2 for those builds.

Comment (s-perron) [reply README.md]: Done I made it explicity 2.7 for linux bots.

**Pull request merged**